gateway/server.go

```diff
@@ -85,8 +85,12 @@ var (
 const appName = "tyk-gateway"
 
 type Gateway struct {
+	startMu sync.Mutex
 	started bool
 
+	checkSumSetMu   sync.Mutex
+	currCheckSumSet map[string]struct{}
```

Comment:
I'm not sure we really need a new map here. Maybe just iterate all specs like previously? Or you found it has significant performance improvement?

Reply:
not really, changed back to use `getApiSpec` instead